src/spec-node/devContainers.ts

```diff
@@ -199,24 +199,40 @@ export interface LogOptions {
 	onDidChangeTerminalDimensions?: Event<LogDimensions>;
 }
 
-export function createLog(options: LogOptions, pkg: PackageConfiguration, sessionStart: Date, disposables: (() => Promise<unknown> | undefined)[], omitHeader?: boolean) {
+export function createLog(options: LogOptions, pkg: PackageConfiguration, sessionStart: Date, disposables: (() => Promise<unknown> | undefined)[], omitHeader?: boolean, secretsP?: Promise<Record<string, string>>) {
 	const header = omitHeader ? undefined : `${pkg.name} ${pkg.version}. Node.js ${process.version}. ${os.platform()} ${os.release()} ${os.arch()}.`;
-	const output = createLogFrom(options, sessionStart, header);
+	const output = createLogFrom(options, sessionStart, header, secretsP);
 	output.dimensions = options.terminalDimensions;
 	output.onDidChangeDimensions = options.onDidChangeTerminalDimensions;
 	disposables.push(() => output.join());
 	return output;
 }
 
-function createLogFrom({ log: write, logLevel, logFormat }: LogOptions, sessionStart: Date, header: string | undefined = undefined): Log & { join(): Promise<void> } {
+function createLogFrom({ log: write, logLevel, logFormat }: LogOptions, sessionStart: Date, header: string | undefined = undefined, secretsP?: Promise<Record<string, string>>): Log & { join(): Promise<void> } {
 	const handler = logFormat === 'json' ? createJSONLog(write, () => logLevel, sessionStart) :
 		process.stdout.isTTY ? createTerminalLog(write, () => logLevel, sessionStart) :
 		createPlainLog(write, () => logLevel);
+	const maskingHelper = async (text: string) => {
+		if (text && secretsP) {
+			return await maskSecrets(secretsP, text);
+		}
+		return text;
+	};
 	const log = {
-		...makeLog(createCombinedLog([handler], header)),
+		...makeLog(createCombinedLog([handler], maskingHelper, header)),
 		join: async () => {
 			// TODO: wait for write() to finish.
 		},
 	};
 	return log;
 }
+
+async function maskSecrets(secretsP: Promise<Record<string, string>>, text: string) {
+	const MASK = '********';
+	const secretValues = Object.values(await secretsP);
+	secretValues.forEach(secret => {
+		const regex = new RegExp(secret, 'g');
```

Comment:
You could create the RexExp once and then reuse it to avoid parsing by `new RegExp()` for each log event. (Left a comment with an example of using a single RegEx, so you only have to keep around one.)